src/ring_theory/ideal/operations.lean

```diff
@@ -903,6 +903,32 @@ begin
     refine ⟨hJ' ▸ map_mono hJ.left, hJ' ▸ map_is_prime_of_surjective hf (le_trans h hJ.left)⟩ },
 end
 
+section quotient_algebra
+
+variable {I : ideal S}
+
+/-- The ring hom `R/f⁻¹(I) →+* S/I` induced by a ring hom `f : R →+* S` -/
+def quotient_map (I : ideal S) (f : R →+* S) : (I.comap f).quotient →+* I.quotient :=
```

Comment:
I think this definition is to restrictive. I would rather see `I : ideal R`, `J : ideal S`, `h : I ≤ J.comap f`, `I.quotient → J.quotient`.
Note that if `I = J.comap f` then `I.quotient` is *not* the same as `(J.comap f).quotient`. This is very unfortunate, but Lean is really picky about this.
In general, even working with `I.quotient` is often to restrictive. We might want to do everything for surjective ring homs instead. (But it is fine to also have stuff for `ideal.quotient`.)

Reply:
I've generalized `quotient_map` (although not all the way to surjective ring homs). I didn't change `algebra_map_quotient` because adding the extra `h` would prevent type class inference of the algebra structure. I also don't think injectivity holds unless `I = J.comap f`, so I left that lemma as is.

Reply:
Yes, `algebra_map_quotient` is fine.